pyproject.toml

```diff
@@ -39,64 +18,66 @@ classifiers = [
   "Topic :: Scientific/Engineering :: Artificial Intelligence",
 ]
 
-[project.optional-dependencies]
-test = [
-    "pytest >= 7.3.1",
-]
-doc = [
+[tool.poetry.dependencies]
+python = ">=3.8.1,<3.12" # updated for llama-index dependency
+black = "23.3.0"
+mypy = "1.3.0"
+openai = "0.28"
+pre-commit = "3.3.3"
+pytest = "7.3.1"
+ruff = ">=0.0.272"
+termcolor = "2.3.0"
+typer = ">=0.3.2"
+rudder-sdk-python = ">=2.0.2"
+dataclasses-json = "0.5.7"
+tiktoken = ">=0.0.4"
+tabulate = "0.9.0"
+python-dotenv = ">=0.21.0"
+langchain = ">=0.0.335"
+llama-index = ">=0.8.49"
+rank-bm25 = ">=0.2.2"
+tree_sitter_languages = ">=1.8.0"
```

Comment:
Can you put these in a new list under [tool.poetry.extras] called something like experimental + relevant readme update?